Sources/Vapor/Utilities/DotEnv.swift

```diff
@@ -25,6 +25,67 @@ import Darwin
 /// Single-quoted strings are parsed literally. Double-quoted strings may contain escaped newlines
 /// that will be converted to actual newlines.
 public struct DotEnvFile {
+    /// Reads the dotenv files relevant to the environment and loads them into the process.
+    ///
+    ///     let environment: Environment
+    ///     let elgp: EventLoopGroupProvider
+    ///     let logger: Logger
+    ///     try DotEnvFile.load(environment: .development, elgp, logger: logger)
+    ///     print(Environment.process.FOO) // BAR
+    ///
+    /// - parameters:
+    ///     - environment: Absolute or relative path of the dotenv file.
+    ///     - eventLoopGroupProvider: Either provides an EventLoopGroup or tells the function to create a new one.
+    ///     - logger: Optionally provide an existing logger.
+    public static func load(
+        environment: Environment = .development,
+        _ eventLoopGroupProvider: Application.EventLoopGroupProvider = .createNew,
```

Comment:
I would change the labels here to make for a cleaner and more readable callsite.
```suggestion
        for environment: Environment = .development,
        on eventLoopGroupProvider: Application.EventLoopGroupProvider = .createNew,
```

Reply:
regarding the nit, 

Maybe we should match the application initializers style? 

```swift
public init(
        _ environment: Environment = .development,
        _ eventLoopGroupProvider: EventLoopGroupProvider = .createNew
) {
```

Reply:
Regarding test, there are tests, they test the individual loading functions.

Reply:
Tests/VaporTests/DotEnvTests.swift

Reply:
I would not go with no labels since this is a static method, and not an init. `Application(.development, .shared(...))` read a lot nicer to me than `DotEnvFile.load(.development, .shared(...), myLogger)`

For the tests, could you duplicate the first one but instead use the new `DotEnvFile.load` and check values in `Process.environment`? :)

Reply:
Sure!

Reply:
Regarding the test, not sure how to handle the current location of the test.env file as it's looking for these in the working directory

Reply:
I could also take the file names out of the function and have the user provide it to the function? That would also fix the testability issue and make the function more generally usable

Reply:
so then it would become something like (then we don't need the environment logic inside the function anymore)
```swift
let fileNames = [".env", ".test.env"]
DotEnvFile.loadFiles(fileNames, on: eventLoopGroupProvider, logger: Logger())
```

Reply:
Let me know what you think @MrLotU

Reply:
Or per file name of course to prevent the use of the map, feels like a bit of double setting up and tearing down the threadPool and eventLoopGroup, but it's cleaner api wise ofc

Reply:
Ah, forgot about the fact you can't specify the path here. I think it's fine as is in that case